bot/exts/moderation/modpings.py

```diff
@@ -116,16 +116,25 @@ async def add_role_schedule(self, mod: Member, work_time: int, schedule_start: d
             log.trace(f"Skipping adding moderator role to mod with ID {mod.id} - found in pings off cache.")
         else:
             log.trace(f"Applying moderator role to mod with ID {mod.id}")
-            await mod.add_roles(self.moderators_role, reason="Moderator scheduled time started!")
+            await self.reapply_role(mod, reason="Moderator scheduled time started!")
 
         log.trace(f"Sleeping for {work_time} seconds, worktime for mod with ID {mod.id}")
         await asyncio.sleep(work_time)
         await self.remove_role_schedule(mod, work_time, schedule_start)
 
-    async def reapply_role(self, mod: Member) -> None:
+    async def reapply_role(self, mod: Member, *, reason: str = None) -> None:
         """Reapply the moderator's role to the given moderator."""
+        if self.guild.get_role(Roles.mod_team) not in mod.roles:
```

Comment:
As far as I can see you have no guarantee that `self.guild` was already set. A possible solution is to set the value in `reschedule_modpings_schedule` as in `reschedule_roles`.

Reply:
The other option is to go with the async init_cog pattern we have in a few other places, where we create a task to init the cog within the cog init and then await that task in all the functions that rely on things like this.

Reply:
e1a580d6